src/Middleware/Rewrite/src/ApacheModRewrite/RuleRegexParser.cs

```diff
@@ -1,11 +1,13 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics.CodeAnalysis;
+
 namespace Microsoft.AspNetCore.Rewrite.ApacheModRewrite;
 
 internal class RuleRegexParser
 {
-    public static ParsedModRewriteInput ParseRuleRegex(string regex)
+    public static ParsedModRewriteInput ParseRuleRegex([StringSyntax(StringSyntaxAttribute.Regex)] string regex)
```

Comment:
This is an internal class.  Does it matter whether this is attributed?  It would only affect a developer working on this library and passing in a literal string.  No existing call sites take literal strings.  I'd revert this and all other similar cases.

Reply:
It's pretty obvious from the name what this is, and there's zero functional benefit if the input isn't a literal.  From my perspective, attributing such things is just noise.